### DISCUSSION TITLE
Please reconsider renaming Remix to React Router.

### DISCUSSION THREAD

**Discussion by boblauer** [remix-run/remix#9452] [General]
I know there has been a lot of blowback on Twitter about Remix v3 becoming React Router v7, but I wanted to open a discussion around the topic because I think it's an important discussion to have.

It just seems so painfully obvious that this rename is a mistake. By dropping the Remix name, you're losing all of the good will and reputation that the Remix brand has gained over the last few years. And on the flip side, by adopting React Router you're taking on all of the legacy baggage that comes with React Router.

You're also making it so much more difficult for developers to find the correct documentation. If I search on SO for a remix issue, I don't have to worry all that much about the answers being irrelevant to the version of Remix that I'm on because Remix hasn't been around that long and it hasn't changed all that much. But the same is not true with React Router. The internet is littered with articles about React Router that are no longer applicable to v6 or v7, but that's not always obvious to the reader.

The most frustrating aspect of this is that this is the absolute perfect opportunity to go the other direction and shed the baggage that comes with React Router. It's very frustrating to see it potentially squandered.

I hope this doesn't come across as an attack. I genuinely love Remix and I want to see it continue to grow and thrive, and I think branding is a big part of any framework's success.

**Comment by gustavopch** [comment]
IMO:
- Making people move from React Router to Remix would be much harder than making them upgrade to the next major.
- By having the popularity of React Router, "Remix" will improve much faster simply because of the number of brains using and working on it.
- Deciding any of this based on how search engines currently work is probably a mistake given web search is about to get much smarter with LLMs. And before that happens, you can limit your search to "Last month", "Last year", etc.

**Comment by bluefire2121** [comment]
IMO:

- React Router represents an older, dated brand for SPA routes with ties to the abandoned CRA.
- Remix represents a trusted web/server architecture: a pure pattern framework for building complex, yet performant web applications.

**Comment by SteveSuv** [comment]
I suggest changing `remix` to `remix.js` to make it easier for browsers, AI, and reddit to search keywords, and to distinguish it from some music terms.

It is not necessary to completely migrate the code of `remix` to `rr`, but to use `rr` as a dependency of `remix`.

Although `rr` has many stars, its influence and reputation are not as good as the official thinks. Many companies or individuals use v4 or earlier versions. And `remix` represents the unified, omnipotent, high DX full-stack framework that front-end developers pursue, maybe should continue to invest `remix`. it's becoming more and more better. Especially after using `vite` this year, `remix` was praised by more people. 

`Remix` represents a larger concept than just routing, it represents the future.

**Reply by kiliman** [reply]
They've already said that Remix won't be going away. The core functionality of Remix and React Router overlaps so much that it didn't make sense to keep them separate. Once the merger is complete, they'll focus on what makes Remix, the "framework," special again.

**Reply by jhanink** [reply]
we shall see.
